lib/Optimizer/Quantization.cpp

```diff
@@ -22,16 +22,23 @@
 
 using namespace glow;
 
-/// Instrument function \p F with Quantization Profile nodes.
-/// Nodes should only be added when function is compiled in
-/// the Inference mode.
-void glow::profileQuantization(Function *F) {
+Function *glow::profileQuantization(Function *F, llvm::StringRef newFuncName) {
+  // Create a new name for the differentiated function, if none is given.
+  std::string tmpName;
+  if (newFuncName.empty()) {
+    tmpName = std::string(F->getName()) + "_profile";
+    newFuncName = tmpName;
```

Comment:
Why not just 

`newFuncName = std::string(F->getName()) + "_profile";`

Reply:
newFuncName is a stringRef. You can't point it to temporary storage. We could copy the name into tmpName, but at this point the code will be almost the same length. What do you think?

Reply:
newFuncName is a StringRef and does not own the pointer to the actual data.
Lifetime of the object created "std::string(F->getName()) + "_profile"" will be shorter than we'd need. That temp object is going to be destroyed right away and newFunName won't be valid.

Reply:
Ah got it. I think it's maybe a bit cleaner to do something like:
```
const std::string finalNewName =  newFuncName.empty()
                                     ? std::string(F->getName()) + "_profile"
                                     : newFuncName
```
But no big deal either way.